### PR TITLE
Improved responsiveness of panning and rotation

### PR DESCRIPTION
**Pull request by Helveg**
Rotation now scales with mouse speed, and panning with zoom level.

Ideally these hardcoded sensitivities could become settings somewhere, for now I tuned them on my machine so that rotation felt good, but this might depend a lot on mouse sensitivity and preference. For the panning I had a better metric, I tuned it so that on all zoom levels the CV markers would stick to the mouse as close as possible during panning.

### CODE REVIEW

File: src/window.cpp [hunk 1]
```diff
@@ -35,16 +37,16 @@ static void mouse_callback(GLFWwindow* window, double x, double y) {
     auto dy = mouse_y - y; mouse_y = y;
 
     if (lb_down && ctrl_key) {
-        delta_pos.x = (std::abs(dx) > eps) ? -dx*2 : 0.0f;
-        delta_pos.y = (std::abs(dy) > eps) ?  dy*2 : 0.0f;
+        delta_pos.x = - dx * pow(gui_state_zoom, 1.2) / 12;
```
Comment (thorstenhater) [inline src/window.cpp:40]: How did you arrive at this formula?

File: src/window.cpp [hunk 2]
```diff
@@ -10,10 +11,11 @@ static void glfw_error_callback(int error, const char* description) {
     log_error("Glfw error {}:\n{}", error, description);
 }
 
+glm::vec2 delta_pos = {0.0f, 0.0f};
 float delta_phi     = 0.0f;
-float delta_gamma     = 0.0f;
+float delta_gamma   = 0.0f;
 float delta_zoom    = 0.0f;
-glm::vec2 delta_pos = {0.0f, 0.0f};
+extern float gui_state_zoom;
```
Comment (thorstenhater) [inline src/window.cpp:18]: I would prefer having less global variables rather than more.

**Review by thorstenhater — CHANGES_REQUESTED** [review]
Thanks, however, in general I am wary of tweaking things to a specific setup. Could you add settings for mouse sensitivity in the menus?

**Comment by Helveg** [comment]
Sure, I'll see if I can find directions in Dear ImGui's docs or bother you on Slack ;p

**Comment by thorstenhater** [comment]
Hi @Helveg,

I am gearing up towards formally releasing this and moving the project over to the arbor-sim org.
This means that by the end of next week this repo will be archived and I will likely close all resolved
PRs by then.

T

**Comment by Helveg** [comment]
That's fine, I don't have the bandwidth to implement the requested changes, feel free to close :)

**Comment by thorstenhater** [comment]
Ok, thanks for the reply. I also overhauled the navigation in a branch that will be merged into the release.

**Pull request closed without merging**